spot_driver/src/robot_state/spot_robot_state_middleware_handle.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver/robot_state/spot_robot_state_middleware_handle.hpp>
+
+namespace {
+constexpr auto kPublisherHistoryDepth = 1;
+constexpr auto kNodeName{"spot_robot_state_publisher"};
+
+// ROS topic names for Spot's robot state publisher
+constexpr auto kJointStatesTopic{"joint_states"};
+constexpr auto kOdomTwistTopic{"odometry/twist"};
+constexpr auto kOdomTopic{"odometry"};
+constexpr auto kFeetTopic{"status/feet"};
+constexpr auto kEStopTopic{"status/estop"};
+constexpr auto kWifiTopic{"status/wifi"};
+constexpr auto kBatteryStatesTopic{"status/battery_states"};
+constexpr auto kPowerStatesTopic{"status/power_states"};
+constexpr auto kSystemFaultsTopic{"status/system_faults"};
+constexpr auto kBehaviorFaultsTopic{"status/behavior_faults"};
+constexpr auto kEndEffectorForceTopic{"status/end_effector_force"};
+constexpr auto kManipulatorTopic{"manipulation_state"};
+}  // namespace
+
+namespace spot_ros2 {
+
+SpotRobotStateMiddlewareHandle::SpotRobotStateMiddlewareHandle(const std::shared_ptr<rclcpp::Node>& node)
+    : node_{node},
+      parameter_interface_{std::make_unique<RclcppParameterInterface>(node)},
```

Comment:
The parameter interface is not used directly by this class, but it is used by the class which owns this class. Therefore, it is a dependency not for this class but for the owning class. We should move it out.

Reply:
I refactored this class to move these objects into the the StatePublisher. I'm thinking about going further with that and splitting up the middleware handle into individual publishers, which would allow testing the conditionals in the middleware handle implementation in more detail.